sensing/tier4_pcl_extensions/include/tier4_pcl_extensions/voxel_grid_nearest_centroid_impl.hpp

```diff
@@ -320,19 +323,15 @@ void pcl::VoxelGridNearestCentroid<PointT>::applyFilter(PointCloud & output)
 
       output.push_back(PointT());
 
-      // TODO(kenji.miyake) refactor
-      double dis_min = 1000000.0;
-      size_t i_min = 0;
-      for (size_t i = 0; i < leaf.points.size(); ++i) {
-        double dis = std::pow(leaf.points.at(i).x - leaf.centroid[0], 2.0) +
-                     std::pow(leaf.points.at(i).y - leaf.centroid[1], 2.0) +
-                     std::pow(leaf.points.at(i).z - leaf.centroid[2], 2.0);
-        if (dis < dis_min) {
-          dis_min = dis;
-          i_min = i;
-        }
-      }
-      output.points.back() = leaf.points.at(i_min);
+      const auto & centroid = leaf.centroid;
+      const auto distances = leaf.points | ranges::views::transform([&centroid](const auto & p) {
+                               return std::pow(p.x - centroid[0], 2) +
+                                      std::pow(p.y - centroid[1], 2) +
+                                      std::pow(p.z - centroid[2], 2);
```

Comment:
If the number of iteration is large, it may be good not to use pow :thinking: 
https://jp.quora.com/C%E3%81%AB%E3%81%8A%E3%81%84%E3%81%A6-%E5%A4%89%E6%95%B0-x-%E3%82%92%E4%BA%8C%E4%B9%97%E3%81%99%E3%82%8B%E5%A0%B4%E5%90%88-pow-x-2-0-%E3%82%88%E3%82%8A-x-x-%E3%81%8C%E6%9C%9B%E3%81%BE%E3%81%97%E3%81%84%E3%81%AE

Reply:
sure, its better not to use std::pow for point cloud iteration... :pleading_face: 
fixed db01af7.